app/controllers/GoodsRecordsController.scala

```diff
@@ -102,25 +102,29 @@ class GoodsRecordsController @Inject() (
 
   def getDownloadLinkMessagesKey(opt: Option[DownloadDataSummary]): String =
     opt match {
-      case Some(downloadDataSummary) if downloadDataSummary.status == RequestFile    =>
+      case Some(downloadDataSummary) if downloadDataSummary.status == RequestFile     =>
         "goodsRecords.downloadLinkText.requestFile"
-      case Some(downloadDataSummary) if downloadDataSummary.status == FileInProgress =>
+      case Some(downloadDataSummary) if downloadDataSummary.status == FileInProgress  =>
         "goodsRecords.downloadLinkText.fileInProgress"
-      case Some(downloadDataSummary) if downloadDataSummary.status == FileReady      =>
+      case Some(downloadDataSummary) if downloadDataSummary.status == FileReadyUnseen =>
         "goodsRecords.downloadLinkText.fileReady"
-      case _                                                                         =>
+      case Some(downloadDataSummary) if downloadDataSummary.status == FileReadySeen   =>
+        "goodsRecords.downloadLinkText.fileReady"
+      case _                                                                          =>
```

Comment:
should this be the case when the option is empty?

Reply:
If the option is empty the have never requested a file so they are at the 'RequestFile' point in the journey